### PR TITLE
fix(TFD-2023): Add required fields to JSON Schema for PropertyLists.

### PR DESCRIPTION
**Pull request by RyanSkraba**
**What is the problem this Pull Request is trying to solve?**

Required fields inside PropertiesLists are not correctly written into JSON Schema.

**What is the chosen solution to this problem?**

Required fields inside PropertiesLists are correctly written into JSON Schema.
 
**Link to the JIRA issue**

https://jira.talendforge.org/browse/TFD-2023
 
**Please check if the Pull Request fulfills these requirements**
- [X] The PR commit message follows our [guidelines](../CONTRIBUTING.md)
- [X] Tests for the changes have been added (for bug fixes / features, coverage should be over 75% in the new code)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] Related design / discussions / pages (not in JIRA), if any, are all linked or available in the Pull Request

### CODE REVIEW

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/daikon_pull-request_master/227/) [:star:](https://ci-common.datapwn.com/job/daikon_pull-request_master/227/)

File: daikon/src/main/java/org/talend/daikon/serialize/jsonschema/JsonSchemaGenerator.java [hunk 1]
```diff
@@ -103,8 +104,12 @@ private void traverseNestedProperties(Properties cProperties, ObjectNode schema,
                 boolean isVisible = widget != null && widget.isVisible();
                 // compute the formName is one of the properties form was added as a subform
                 String propertiesFormName = null;
-                if (isVisible && widget.getContent() instanceof Form) {
-                    propertiesFormName = widget.getContent().getName();
+                if (isVisible) {
+                    if (widget.getContent() instanceof Form) {
+                        propertiesFormName = widget.getContent().getName();
+                    } else if (properties instanceof PropertiesList) {
+                        propertiesFormName = properties.getForm(null).getName();
```
Comment (lbourgeois) [inline daikon/src/main/java/org/talend/daikon/serialize/jsonschema/JsonSchemaGenerator.java]: This change is required so the _addToRequired_ method is called right ?

File: daikon/src/main/java/org/talend/daikon/serialize/jsonschema/JsonSchemaGenerator.java [hunk 2]
```diff
@@ -70,6 +70,7 @@ private void computeSchemaType(Properties cProperties, String formName, boolean
 
         // Handle PropertiesList type
         if (cProperties instanceof PropertiesList<?>) {
+            schema.put(JsonSchemaConstants.TAG_TITLE, cProperties.getDisplayName());
```
Comment (lbourgeois) [inline daikon/src/main/java/org/talend/daikon/serialize/jsonschema/JsonSchemaGenerator.java]: What is this change for ?

Comment (RyanSkraba) [reply daikon/src/main/java/org/talend/daikon/serialize/jsonschema/JsonSchemaGenerator.java]: Added comment!

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/daikon_pull-request_master/228/) [:star:](https://ci-common.datapwn.com/job/daikon_pull-request_master/228/)

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/daikon_pull-request_master/229/) [:star:](https://ci-common.datapwn.com/job/daikon_pull-request_master/229/)

**Pull request merged**